pucas/ldap.py

```diff
@@ -84,9 +84,30 @@ def user_info_from_ldap(user):
     user_info = LDAPSearch().find_user(user.username)
     if user_info:
         for user_attr, ldap_attr in attr_map.items():
-            # NOTE: is it possible some values are not set at all here,
-            # in which case we'd get 'None' string values set?
-            setattr(user, user_attr, str(getattr(user_info, ldap_attr)))
+            # Handle issues where an attribute may need to be populated by
+            # multiple attributes OR where it is missing, in which case
+            # handle the LDAPCursorError and set None to empty string.
+
+            # if the user passed a list, try in order until we get the attr
+            # or we fail to do so.
+            if isinstance(ldap_attr, list):
+                for val in ldap_attr:
+                    try:
+                        setattr(user, user_attr, str(getattr(user_info, val)))
+                        break
+                    except LDAPCursorError:
+                        # NOTE: it would be more efficient to do this as a single
```

Comment:
I'm not following this note. Would using a `MagicMock` solve the testing issue you're having?

Reply:
Sadly no. The issue is that `MagicMock` and `Mock` both don't let you adjust the behavior of their `__getattr__` MagicMethod, so any check for it being `None` or falsy fails. However, I believe your suggestion solves the problem in general.

Reply:
@bwhicks instead of overriding `getattr`, can you just set attributes?